Source/WebCore/rendering/svg/SVGRenderSupport.cpp

```diff
@@ -267,7 +267,7 @@ void SVGRenderSupport::layoutChildren(RenderElement& start, bool selfNeedsLayout
                     svgText.setNeedsTextMetricsUpdate();
                     svgText.setNeedsPositioningValuesUpdate();
                 }
-
+                child.setNeedsTransformUpdate();
```

Comment:
What are the performance implications of always calling this?

Reply:
this is only happening when the RenderSVGRoot changes size.

Reply:
Eww, be careful, if the RenderSVGRoot size changes, you are now seting this flag on a all nodes of certain types (e.g. all <rect> elemtns etc.) That is because "element.hasRelativeLengths()" unconditionally returns true here:
```
svg/SVGCircleElement.h:    bool selfHasRelativeLengths() const final { return true; }
svg/SVGEllipseElement.h:    bool selfHasRelativeLengths() const final { return true; }
svg/SVGFilterElement.h:    bool selfHasRelativeLengths() const final { return true; }
svg/SVGForeignObjectElement.h:    bool selfHasRelativeLengths() const final { return true; }
svg/SVGImageElement.h:    bool selfHasRelativeLengths() const final { return true; }
svg/SVGMaskElement.h:    bool selfHasRelativeLengths() const final { return true; }
svg/SVGPatternElement.h:    bool selfHasRelativeLengths() const final { return true; }
svg/SVGRectElement.h:    bool selfHasRelativeLengths() const final { return true; }
```

That was changed in 2014 by Dirk (see 6d66047e8a4c9), see
```
    Since we turned width and height to presentation attributes, the layout optimization
    selfHasRelativeLengths() in the DOM can't be used anymore. selfHasRelativeLengths() was
    intended to solve a problem where we did not layout relatively position/sized elements
    when the parent changes its size. However, as a side effect it did not call layout
    for absolutely positioned/sized elements since the layout does not change. I run
    all performance tests that we have and even wrote a test with hundreds of elements
    that would be affected by this optimization. The differences were inside the sigma
    of a normal test run. (Means I couldn't measure a performance difference.)
```

IMHO the conclusion is wrong though, since width/height are presentation attributes, selfHasRelativeLengths() mechanism should be unified/combined with the possibility to check the CSSValues for percentage units -- because we only ever want to layout() a SVG renderer upon window size changes, if it acually depends on the window size, everything else is an excuse. And I also doubt that this is within one sigma in general -- I could construct a pathological case, where the transform recomputation overhead gets significant (one needs many transformed elements with absolute units, that usually wouldn't need layout, but do because of the selfHasRelativeLengths()==true hack).

I would strongly recommend to measure the impact of the patch and eventually fix the relative length mechanism to consider CSS properties too.